File: c++ programs/spread_the_word.cpp
#include <bits/stdc++.h> 
using namespace std; 

int main() 
{ 
    long long int t, total_people, day_count, people_know, i, num;
    cin >> t;
    while(t--)
    {
        cin >> total_people;
        long long int a[total_people];
        for(i = 0; i<total_people; ++i)
        {
            cin >> a[i];
        }

        day_count = 0;
        people_know = 1;
        while(people_know < total_people)
        {
            num = people_know;
            for(i=0; i < num; ++i)
                people_know = people_know + a[i];
            day_count++;
        }
        cout << day_count << "\n"; 
    }
    return 0;
}

File: python programs/spread_the_word.py
t = int(input())
for i in range(t):
	num = int(input())
	l2 = list(map(lambda x : int(x), input().split()))
	day_count = 0
	people_know = 1
	total_people = len(l2)
	while(people_know < total_people):
		for i in range(people_know):
			people_know += l2[i]
		day_count += 1
	print(day_count)